### PR TITLE
Add JSX components to blog-core

### PR DESCRIPTION
**Pull request by laurieontech**
Add only the components so they can be leveraged. No Layout, footer, header extras.

### CODE REVIEW

File: packages/gatsby-theme-blog-core/src/components/post-hero.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from "react"
+import Image from "gatsby-image"
+
+const Hero = ({ post }) => (
+  <>
```
Comment (jxnblk) [inline packages/gatsby-theme-blog-core/src/components/post-hero.js:5]: NB: I don't think the fragment is necessary here, if the next line resolves to false, it shouldn't render anything

File: packages/gatsby-theme-blog-core/src/components/post.js [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+import PostTitle from "./post-title"
+import PostDate from "./post-date"
+import PostHero from "./post-hero"
+
+const Post = ({data}) => {
+    const post = data.blogPost
+    return (
+        <main>
+        <PostHero post={post} />
+        <PostTitle>{post.title}</PostTitle>
```
Comment (jxnblk) [inline packages/gatsby-theme-blog-core/src/components/post.js:14]: I'm guessing the reason for not keeping this markup structure all in the same place is to allow for finer-grained shadowing, right?

File: packages/gatsby-theme-blog-core/src/components/posts.js [hunk 3]
```diff
@@ -1,3 +1,14 @@
 import React from "react"
 
-export default props => <pre>{JSON.stringify(props.data, null, 2)}</pre>
+import PostList from "./post-list"
+
+const Posts = ({ data }) => { 
+    const posts = data.allBlogPost.edges
+    return (
+        <main>
```
Comment (jxnblk) [inline packages/gatsby-theme-blog-core/src/components/posts.js:8]: Side note: we should make sure prettier & lint-staged are set up in this monorepo cc @johno

File: packages/gatsby-theme-blog-core/src/components/post.js (hunk 2 above)
Comment (laurieontech) [reply packages/gatsby-theme-blog-core/src/components/post.js:14]: Ya, that was the idea. And to match the structure in blog theme. Arguably we could collapse it. I can see both sides.

File: packages/gatsby-theme-blog-core/src/components/post-hero.js (hunk 1 above)
Comment (laurieontech) [reply packages/gatsby-theme-blog-core/src/components/post-hero.js:5]: React doesn't seem to link it without the fragment. I was going to dig into why, but seemed like it wasn't worth it for now.

**Pull request merged**